File: src/main/java/ru/dkolmogortsev/customui/DailyGridContainer.java
package ru.dkolmogortsev.customui;

import javafx.scene.layout.FlowPane;

/**
 * Created by dkolmogortsev on 3/11/17.
 */
public class DailyGridContainer extends FlowPane
{
    public DailyGridContainer()
    {
        setVgap(10);
    }

    public int getGridIndex(long date)
    {
        for (int i = 0; i < getChildren().size(); i++)
        {
            DayGridPane dayGrid = (DayGridPane)getChildren().get(i);
            if (dayGrid.getDate() == date)
            {
                return i;
            }
        }
        return -1;
    }

}

File: src/main/kotlin/ru/dkolmogortsev/customui/DailyGridContainer.kt
package ru.dkolmogortsev.customui

import javafx.scene.layout.FlowPane

/**
 * Created by dkolmogortsev on 3/11/17.
 */
class DailyGridContainer : FlowPane() {
    init {
        vgap = 10.0
    }

    fun getGridIndex(date: Long): Int {
        for (i in 0..children.size - 1) {
            val dayGrid = children[i] as DayGridPane
            if (dayGrid.date == date) {
                return i
            }
        }
        return -1
    }

}
